src/backend/models/flatArray.ts

```diff
@@ -0,0 +1,158 @@
+import Page from './page';
+import PageOrder from './pageOrder';
+import NodeCache from 'node-cache';
+
+// Create cache for flat array
+const cache = new NodeCache({ stdTTL: 120 });
+
+/**
+ * @typedef {object} FlatArrayData
+ * @property {string} id - page id
+ * @property {string} parentId - page parent id
+ * @property {string} rootId - page root id
+ * @property {number} level - page level in sidebar
+ * @property {string} title - page title
+ * @property {string} uri - page uri
+ */
+export interface FlatArrayData {
+  id?: string;
+  parentId?: string;
+  rootId?: string;
+  level?: number;
+  title?: string;
+  uri?: string;
+}
+
+/**
+ * @class FlatArray
+ * @class FlatArray model
```

Comment:
useless description. It's better to describe the reason of why this model exists